modules/apps/transfer/keeper/forwarding.go

```diff
@@ -49,7 +49,11 @@ func (k Keeper) acknowledgeForwardedPacket(ctx sdk.Context, packet channeltypes.
 		return errorsmod.Wrap(channeltypes.ErrChannelCapabilityNotFound, "module does not own channel capability")
 	}
 
-	return k.ics4Wrapper.WriteAcknowledgement(ctx, capability, packet, ack)
+	if err := k.ics4Wrapper.WriteAcknowledgement(ctx, capability, packet, ack); err != nil {
+		return err
+	}
+
+	return k.deleteForwardedPacket(ctx, packet.SourcePort, packet.SourceChannel, packet.Sequence)
```

Comment:
Could we add checks in our testing for deletion of the right packet?

Reply:
Done for the `OnAcknowledgment`, for the timeout case I have #6596 in review and I'll ad it to that test once we merge this.